tests/lib/test_config.py

```diff
@@ -1,176 +1,206 @@
-
-    assert err.value.errors[0].path == "CONFIG.simple_key"
-    assert err.value.errors[0].message == "Referenced variable should resolved to a string"
-
-    assert err.value.errors[1].path == "CONFIG.array[2]"
-    assert err.value.errors[1].message == "Referenced variable should resolved to a string"
-
-    assert err.value.errors[2].path == "CONFIG.dict.b"
-    assert err.value.errors[2].message == "Referenced variable should resolved to a string"
-
-    assert err.value.errors[3].path == "CONFIG.deep_dict.sub_dict.b"
-    assert err.value.errors[3].message == "Referenced variable should resolved to a string"
-
-    assert err.value.errors[4].path == "CONFIG.deep_dict.sub_array[0].a"
-    assert err.value.errors[4].message == "Referenced variable should resolved to a string"
 
-    assert err.value.errors[5].path == "CONFIG.deep_dict.sub_array[1].b"
-    assert err.value.errors[5].message == "Referenced variable should resolved to a string"
+    # pylint: disable=unused-argument
+    def test_resolve_variables_deep_with_invalid_reference(self, io_mock):
+        with self.assertRaises(AggregatedConfigurationError) as context:
+            # pylint: disable=protected-access
+            config._resolve_variables_deep(
+                {
+                    "error": {},
+                    "simple_key": "__{{error}}__",
+                    "array": ["0", "1", "2__{{error}}__", ],
+                    "dict": {"a": "val_a", "b": "{{error}}", },
+                    "deep_dict": {
+                        "sub_dict": {"a": "val_a", "b": "{{error}}"},
+                        "sub_array": [
+                            {"a": "{{error}}", "b": "val_b"},
+                            {"a": "val_a", "b": "{{error}}"},
+                        ],
+                    },
+                }
+            )
+
+            err = context.exception
+            assert str(err.value) == "Invalid configuration"
+
+            assert err.value.errors[0].path == "CONFIG.simple_key"
+            assert err.value.errors[0].message == "Referenced variable should resolved to a string"
+
+            assert err.value.errors[1].path == "CONFIG.array[2]"
+            assert err.value.errors[1].message == "Referenced variable should resolved to a string"
+
+            assert err.value.errors[2].path == "CONFIG.dict.b"
+            assert err.value.errors[2].message == "Referenced variable should resolved to a string"
+
+            assert err.value.errors[3].path == "CONFIG.deep_dict.sub_dict.b"
+            assert err.value.errors[3].message == "Referenced variable should resolved to a string"
+
+            assert err.value.errors[4].path == "CONFIG.deep_dict.sub_array[0].a"
+            assert err.value.errors[4].message == "Referenced variable should resolved to a string"
+
+            assert err.value.errors[5].path == "CONFIG.deep_dict.sub_array[1].b"
+            assert err.value.errors[5].message == "Referenced variable should resolved to a string"
```

Comment:
I know that it was already done this way but couldn't we test the whole array instead of testing each item?